src/app/services/http/model-3d-animal-http-service/model-3d-animal-http.service.ts

```diff
@@ -0,0 +1,191 @@
+        this.search = { ...this.search, ...meta };
+      },
+      setTotalItems: (totalPages) => {
+        this.totalItems = createPageArray(totalPages);
+      },
+      context: `${this.constructor.name}#getAll`,
+    });
+  }
+
+  /**
+   * Get from database 3D model by animal ID and update signal
+   * @param animalId ID of the animal
+   * @author nav
+   */
+   getByAnimalId(animalId: number): void {
+    const url = `${this.sourceUrl}/animal/${animalId}`;
+    this.http.get<IResponse<Model3DAnimal>>(url).subscribe({
+      next: (res) => {
+        if (res.data && res.data.urlModelo) {
+          this.currentModel3D.set(res.data);
+        } else {
+          this.currentModel3D.set(null);
+        }
+      },
+      error: (error) => {        
+        this.alertService.displayAlert({
+          type: AlertTypeEnum.ERROR,
+          messageKey: `Error cargando modelo 3D: ${error.status === 500 ? 'Error del servidor' : 'Error de conexión'}`,
+        });
+        this.currentModel3D.set(null);
+        this.handleError({
+          message: 'Error loading 3D model for animal',
+          context: `${this.constructor.name}#getByAnimalId`,
+        })(error);
+      }
+    });
+  }
+  /**
+   * Get 3D model by ID and update signal
+   * @param id ID of the 3D model
+   * @author nav
+   */
+  getById(id: number): void {
+    this.find(id).subscribe({
+      next: (res) => {
+        this.currentModel3D.set(res.data);
+      },
+      error: this.handleError({
+        message: 'Error loading 3D model',
+        context: `${this.constructor.name}#getById`,
+      })
+    });
+  }
+
+  /**
+   * Sends a new 3D model to the server for creation.
+   * On success, shows a success alert and refreshes the paginated list.
+   * On failure, displays an error alert and logs the error.
+   * 
+   * @param model3D The 3D model object to be created.
+   * @returns void
+   * @author nav
+   */
+  save(model3D: Model3DAnimal): void {
+    this.add(model3D).subscribe({
+      next: (response) => {
+        this.alertService.displayAlert({
+          type: AlertTypeEnum.SUCCESS,
+          messageKey: response.message,
+        });
+        this.getAll();
+      },
+      error: this.handleError({
+        message: 'An error occurred adding the 3D model',
+        context: `${this.constructor.name}#save`,
+      }),
+    });
+  }
+
+  /**
+   * Updates an existing 3D model by its ID.
+   * If the ID is not present, the update is aborted and an error alert is shown.
+   * On success, shows a success alert and refreshes the paginated list.
+   * On failure, displays an error alert and logs the error.
+   * 
+   * @param model3D The 3D model object with updated fields. Must contain a valid `id`.
+   * @returns void
+   * @author nav
+   */
+  update(model3D: Model3DAnimal): void {
+    if (!model3D.id) {
+      this.alertService.displayAlert({
+        type: AlertTypeEnum.ERROR,
+        messageKey: '3D Model ID is required for update',
```

Comment:
Este valor del objeto recibe una llave de traduccion si le pasas esto ah asi en la interfaz va a aparecer algo como 

```typescript
"???  3D Model ID is required for update ???"
```

si querias usar el que no recibe la llave de i18n creo que es mas bien ah si. Tendrias que confirmarlo si se llama ah si "message"

```suggestion
        message '3D Model ID is required for update',
```